Backpack/src/main/java/net/skyscanner/backpack/calendar/presenter/BpkCalendarController.kt

```diff
@@ -55,6 +55,9 @@ abstract class BpkCalendarController {
     onRangeSelected(selectedRange)
   }
 
+  // TODO: This will not respect differences in the order of the fields for each country.
+  // E.g. US = mm/dd/yyyy and UK = dd/mm/yyyy
+  // Shall we use DateFormat.getDateInstance instead?
```

Comment:
do we plan to fix this in this PR?

Reply:
Not really, it's really a problem as far as I can tell because just now it is only used to localise the weekdays, so it doesn't matter much. But we can improve it at some point

Reply:
Do you mean `SimpleDateFormat(pattern, locale)` is not enough? As they are injected when the Controller is subclassed, it should not be an issue. Locale in our app is generated from our custom set of rules and therefore it should respect the differences of the fields based on the locale that is passed here. Passing on `mm/dd/YYYY` but UK locale should correctly format the date

Reply:
What I meant is that for the UK dates should be `dd/mm/yyyy` and US `mm/dd/yyyy`. Doing `SimpleDateFormat("MM/dd/YYYY", Locale.UK).format(Calendar.getInstance().time)` returns `01/14/2019` which works in the sense that it formats the date but it's not the correct pattern. 

Now if the pattern is always correct then there is no problem. The consumer has to provide the correct pattern, which I believe is what the app does. But there are ways of letting java do it. 

In any case I don't want to change it now, I just had the TODO there so we can revisit at some point and improve it or just leave it as it is.